remote-config/src/test/kotlin/com/rakuten/tech/mobile/remoteconfig/verification/AesEncryptorSpec.kt

```diff
@@ -0,0 +1,83 @@
+package com.rakuten.tech.mobile.remoteconfig.verification
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.anyOrNull
+import com.nhaarman.mockitokotlin2.mock
+import com.rakuten.tech.mobile.remoteconfig.RobolectricBaseSpec
+import org.amshove.kluent.*
+import org.junit.Before
+import org.junit.Test
+import java.security.KeyStore
+import javax.crypto.KeyGenerator
+import javax.crypto.SecretKey
+
+class AesEncryptorSpec : RobolectricBaseSpec() {
+
+    private val mockKeyStore: KeyStore = mock()
+    private val mockKeyGenerator: AesKeyGenerator = mock()
+    private val mockKeyEntry: KeyStore.SecretKeyEntry = mock()
+
+    @Before
+    fun setup() {
+        val key = generateKey()
+        When calling mockKeyGenerator.generateKey() itReturns key
+        When calling mockKeyEntry.secretKey itReturns key
+        When calling mockKeyStore.getEntry(any(), anyOrNull()) itReturns mockKeyEntry
+    }
+
+    @Test
+    fun `it should encrypt the data`() {
+        val encryptor = createAesEncryptor()
+
+        encryptor.encrypt("test data") shouldNotContain "test data"
+    }
+
+    @Test
+    fun `it should attach IV`() {
+        val encryptor = createAesEncryptor()
+
+        encryptor.encrypt("test data") shouldContain "\"iv\":"
+    }
+
+    @Test
+    fun `it should decrypt the data`() {
+        val encryptor = createAesEncryptor()
+
+        val encryptedData = encryptor.encrypt("test data")
+
+        encryptor.decrypt(encryptedData) shouldEqual "test data"
+    }
+
+    @Test
+    fun `it should generate a new key when one is not in the key store`() {
+        When calling mockKeyStore.getEntry(any(), anyOrNull()) itReturns null itReturns mockKeyEntry
```

Comment:
Is that syntax right? Two `itReturns`?

Reply:
Yeah, the the first one is used for the first call and the second for the second call.